### PR TITLE
Fix and improve "create" questions

### PR DESCRIPTION
**Pull request by DenysVuika**
**What kind of change does this PR introduce?**
bugfix + feature

**Did you add tests for your changes?**
No, manual tests only

**Summary**

For the time being, running `preact create` without additional parameters causes the tool to crash because of the bug in the input types for user questions.

This PR introduces the following changes:

- upgrade to latest version of `prompts`
- fix the input types
- handle situation when user cancels the questions (`Ctrl+C`), do not continue

Screenshot 2019-08-17 at 20 41 43

- move the question generator closer to the "create" command code (was in the "setup" previously but used only for "create")
- introduce the template selector

Screenshot 2019-08-16 at 10 13 25

- using `custom` template shows extra question to provide the value
- ask about  package manager (yarn/npm) only if `Install dependencies` selected
- confirm the directory overwrite only when it already exists, abort the application correctly

Screenshot 2019-08-17 at 20 40 48

**Does this PR introduce a breaking change?**
No

**Other information**

Please paste the results of `preact info` here.

### CODE REVIEW

File: packages/cli/lib/commands/create.js [hunk 1]
```diff
@@ -8,27 +8,116 @@ const { resolve, join } = require('path');
...
+			message: 'Directory to create the app',
+		},
+		// Extra data / flags
+		{
+			type: argv.name ? null : 'text',
+			name: 'name',
+			message: 'The name of your application',
+		},
+		{
+			type: argv.force ? null : 'confirm',
```
Comment (ForsakenHarmony) [inline packages/cli/lib/commands/create.js]: should this only prompt if the directory exists?

Comment (DenysVuika) [reply packages/cli/lib/commands/create.js]: That was not in the original code but I can add the check

Comment (DenysVuika) [reply packages/cli/lib/commands/create.js]: added more cases for the question flow, see the PR description

**Comment by DenysVuika** [comment]
@ForsakenHarmony thanks for the review. Does this ticket need anything from me to progress, or it will be merged later?

**Comment by ForsakenHarmony** [comment]
wanted someone else to do a review, but I think it's good, so I'll just merge it now, thanks 🎉

File: packages/cli/lib/commands/create.js [hunk 2]
```diff
@@ -8,27 +8,138 @@ const { resolve, join } = require('path');
...
 const { info, isDir, hasCommand, error, trim, warn } = require('../util');
-const { addScripts, install, initGit, isMissing } = require('../lib/setup');
+const { addScripts, install, initGit } = require('../lib/setup');
 
 const ORG = 'preactjs-templates';
 const RGX = /\.(woff2?|ttf|eot|jpe?g|ico|png|gif|webp|mp4|mov|ogg|webm)(\?.*)?$/i;
 const isMedia = str => RGX.test(str);
 const capitalize = str => str.charAt(0).toUpperCase() + str.substring(1);
 
+function directoryExists(workingDir, destDir) {
```
Comment (prateekbh) [inline packages/cli/lib/commands/create.js]: nit: this should be in `../util`

**Comment by DenysVuika** [comment]
@prateekbh I am working on additional contribution related to CSS preprocessor selection, will move that function to utils there if you don’t mind

**Pull request merged**